include/UpcGenerator.h

```diff
@@ -134,24 +141,58 @@ class UpcGenerator
 
   // number of threads for two-photon luminosity calculation
   void setNumThreads(int n) { numThreads = n; }
+  
+  void setParFile(std::string parfilename) { parFileName = parfilename; }
 
   // ----------------------------------------------------------------------
-
-  // file parser
-  void initGeneratorFromFile();
+  // set beam parameters
+  void setCollisionSystem(float sqrts, int nucl_z, int nucl_a);
+
+  // set seed for random generator
+  void setSeed(int seedIn) {
+    seed = seedIn;
+    
+    // initialize the MT64 random number generator
+    if (!gRandom) {
+      gRandom = new TRandomMT64();
+    } 
```

Comment:
I would suggest to just
```
delete gRandom;
gRandom = new TRandomMT64();
```
since (to my knowledge) ROOT creates default gRandom object, so `!gRandom` would always be `false`